### PR TITLE
Remaining accessibility fixes to the PL site

### PR DESCRIPTION
**Pull request by clrux**
This relatively small bit of work cleans up the remaining (pertinent) accessibility issues.

Some work performed:

* contrast issues on links, especially those on gray
* set background and foregrounds on the `.sr-only` elements (to black and white, respectively)
* added a few `aria-label`'s on landmark elements
* markup and structure improvements

Some errors that aren't relevant:

* form field examples don't have submit buttons; this is okay since these are just examples
* a few of the colors have contrast issues; these are intentionally shown and we note them visually

---

Please review:

- [x] @talbs 
- [x] @cptvitamin

### CODE REVIEW

File: _src/pattern-library/sass/base/_reset.scss [hunk 1]
```diff
@@ -104,6 +104,13 @@ pre {
 // ----------------------------
 .sr-only {
     @extend %a11y-hide;
+    background: black !important; // providing contrast
```
Comment (clrux) [inline _src/pattern-library/sass/base/_reset.scss]: I chose to use `!important` here to block any overrides that occur later. These elements aren't seen anyway, but we need to provide sufficient contrast for automated checkers.

File: _src/pattern-library/sass/utilities/_helpers.scss [hunk 2]
```diff
@@ -122,7 +122,7 @@
     );
 
     border-bottom: 1px solid $transparent;
-    color: palette(primary, base);
+    color: palette(primary, dark);
```
Comment (clrux) [inline _src/pattern-library/sass/utilities/_helpers.scss]: The base color is fine on white, but it barely fails on gray. I darkened this a bit, which is suitable on both white or gray.

File: _src/pldoc/static/sass/_components.scss [hunk 3]
```diff
@@ -262,7 +266,7 @@
 
     .pldoc-copy {
         font-size: font-size(small);
-        color: palette(grayscale, mid-light);
+        color: palette(grayscale, base);
```
Comment (clrux) [inline _src/pldoc/static/sass/_components.scss]: The copyright info. Also, we didn't  have a `mid-light` variable.

File: _includes/navigation.html [hunk 4]
```diff
@@ -1,7 +1,7 @@
-<nav class="pldoc-nav pldoc-nav-elements pldoc-nav-internal" id="pldoc-nav-elements">
-    <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
+<nav class="pldoc-nav pldoc-nav-elements pldoc-nav-internal" id="pldoc-nav-elements" role="navigation" aria-label="Patterns and elements">
+    <h2 class="pldoc-heading pldoc-nav-title hd-6" id="heading-elements">Elements</h2>
```
Comment (talbs) [inline _includes/navigation.html]: Please prefix this with ``pl-``.

File: _includes/header.html [hunk 5]
```diff
@@ -1,17 +1,17 @@
-<header class="pldoc-header-site">
+<header class="pldoc-header-site" role="banner" aria-labelledby="pldoc_header_site_title">
```
Comment (talbs) [inline _includes/header.html]: Please use ``-`` instead of ``_`` in your value naming conventions.

File: _includes/header.html [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
     {% if page.title == "Home" %}
-    <h1 class="pldoc-header-site-title">
+    <h1 class="pldoc-header-site-title" id="pldoc_header_site_title">
```
Comment (talbs) [inline _includes/header.html]: Please use ``-`` instead of ``_`` in your value naming conventions.

File: _includes/header.html [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
         <a class="pldoc-link title-name" href="{{ site.baseurl }}">
             {{ site.title }}
         </a>
-        <span class="title-description copy copy-base">{{ site.tagline }}</span>
     </h1>
+    <span class="title-description copy copy-base">{{ site.tagline }}</span>
```
Comment (talbs) [inline _includes/header.html]: Since this is now outside of the ``pldoc-header-site-title`` element, can you add a prefix of ``pl-`` to the class name?

File: _includes/header.html [hunk 8]
```diff
@@ -1,17 +1,17 @@
...
     {% else %}
     <h2 class="pldoc-header-site-title">
         <a class="pldoc-link title-name" href="{{ site.baseurl }}">
             {{ site.title }}
         </a>
-        <span class="title-description copy copy-base">{{ site.tagline }}</span>
     </h2>
+    <span class="title-description copy copy-base">{{ site.tagline }}</span>
```
Comment (talbs) [inline _includes/header.html]: Since this is now outside of the ``pldoc-header-site-title`` element, can you add a prefix of ``pl-`` to the class name?

File: _includes/header.html [hunk 9]
```diff
@@ -1,17 +1,17 @@
...
     {% endif %}
```
Comment (talbs) [inline _includes/header.html]: Also, your ``<span class="title-description copy copy-base">{{ site.tagline }}</span>`` line can be moved outside of the Jekyll if statement to only include it once (rather than twice in the source code). Please simplify.

File: _includes/navigation.html [hunk 10]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
-    <h2 class="pldoc-heading pldoc-nav-title hd-6">Documentation</h2>
-    <ul class="pldoc-list nav-list list-styleguides">
+<nav class="pldoc-nav pldoc-nav-documentation pldoc-nav-external" id="pldoc-nav-documentation" role="navigation" aria-label="Documentation">
+    <h2 class="pldoc-heading pldoc-nav-title hd-6" id="heading-documentation">Documentation</h2>
+    <ul class="pldoc-list nav-list list-styleguides" aria-labelledby="heading-documentation">
         <li class="nav-list-item">
             <a class="pldoc-link" rel="external" href="https://github.com/edx/ux-pattern-library/wiki/Styleguide:-General">
                 <span class="pldoc-icon" aria-hidden="true">&#65514;</span>
                 General Styleguide
+                <span class="sr-only">Link opens in a new window</span>
```
Comment (talbs) [inline _includes/navigation.html]: Hmmm, is this the general best practice for new tab/window links? If so, please change the copy and wrap this text in parenthesis so the link copy reads "Link Text (this link in a new window)"

File: _includes/navigation.html [hunk 11]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
             </a>
         </li>
         <li class="nav-list-item">
             <a class="pldoc-link" rel="external" href="https://github.com/edx/ux-pattern-library/wiki/Styleguide:-HTML">
                 <span class="pldoc-icon" aria-hidden="true">&#65514;</span>
                 HTMLS Styleguide
+                <span class="sr-only">Link opens in a new window</span>
```
Comment (talbs) [inline _includes/navigation.html]: See [my note above](https://github.com/edx/ux-pattern-library/pull/126/files#r35094696).

File: _includes/navigation.html [hunk 12]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
             </a>
         </li>
         <li class="nav-list-item">
             <a class="pldoc-link" rel="external" href="https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS">
                 <span class="pldoc-icon" aria-hidden="true">&#65514;</span>
                 Sass &amp; CSS Styleguide
+                <span class="sr-only">Link opens in a new window</span>
```
Comment (talbs) [inline _includes/navigation.html]: See [my note above](https://github.com/edx/ux-pattern-library/pull/126/files#r35094696).

File: _includes/navigation.html [hunk 13]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
             </a>
         </li>
         <li class="nav-list-item">
             <a class="pldoc-link" rel="external" href="https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Accessibility">
                 <span class="pldoc-icon" aria-hidden="true">&#65514;</span>
                 Accessibility Styleguide
+                <span class="sr-only">Link opens in a new window</span>
```
Comment (talbs) [inline _includes/navigation.html]: See [my note above](https://github.com/edx/ux-pattern-library/pull/126/files#r35094696).

File: _includes/navigation.html [hunk 14]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
             </a>
         </li>
     </ul>
 </nav>
 
-<nav class="pldoc-nav pldoc-nav-project pldoc-nav-external" id="pldoc-nav-project">
-    <h2 class="pldoc-heading pldoc-nav-title hd-6">About This Project</h2>
+<nav class="pldoc-nav pldoc-nav-project pldoc-nav-external" id="pldoc-nav-project" role="navigation" aria-label="About">
+    <h2 class="pldoc-heading pldoc-nav-title hd-6" id="heading-about">About This Project</h2>
```
Comment (talbs) [inline _includes/navigation.html]: Please prefix this with ``pldoc-``.

File: _includes/navigation.html [hunk 15]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
 
-    <ul class="nav-list list-project">
+    <ul class="nav-list list-project" aria-labelledby="heading-about">
```
Comment (talbs) [inline _includes/navigation.html]: Sync up with the rename request above.

File: _includes/navigation.html [hunk 16]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
         <li class="nav-list-item">
             <a class="pldoc-link" rel="external" href="https://github.com/edx/ux-pattern-library/">
                 <span class="pldoc-icon" aria-hidden="true">&#65514;</span>
                 View on Github
+                <span class="sr-only">Link opens in a new window</span>
```
Comment (talbs) [inline _includes/navigation.html]: See [my note above](https://github.com/edx/ux-pattern-library/pull/126/files#r35094696).

File: _includes/navigation.html [hunk 17]
```diff
@@ -15,50 +15,56 @@ <h2 class="pldoc-heading pldoc-nav-title hd-6">Elements</h2>
...
             </a>
         </li>
         <li class="nav-list-item">
             <a class="pldoc-link" rel="external" href="https://openedx.atlassian.net/browse/UX-1396">
                 <span class="pldoc-icon" aria-hidden="true">&#65514;</span>
                 Project Planning
+                <span class="sr-only">Link opens in a new window</span>
```
Comment (talbs) [inline _includes/navigation.html]: See [my note above](https://github.com/edx/ux-pattern-library/pull/126/files#r35094696).

File: _layouts/default.html [hunk 18]
```diff
@@ -19,7 +19,7 @@
             <hr class="pldoc-divider sr-only">
 
             <div class="pldoc-wrapper-content-main">
-                <main class="pldoc-content pldoc-content-main" id="content-main">
+                <main class="pldoc-content pldoc-content-main" id="content-main" aria-label="Main">
```
Comment (talbs) [inline _layouts/default.html]: Is this needed? I thought the ``<main>`` element already gave assistive tech a ton of Main-ish things by default, including a hint that this was the main landmark of a view.

File: _posts/elements/2015-05-12-forms.md [hunk 19]
```diff
@@ -38,8 +38,8 @@ info: Forms allow users to interact with the interface, set preferences, and ent
             </div>
             <div class="field">
                 <label class="field-label field-required" for="email_address_default_helper">Email address (default, required text field with helper text)</label>
-                <input class="field-input input-text" type="email" id="email_address_default_helper" name="email_address_default_helper" placeholder="e.g. jane@email.com" aria-describedby="email_address_default_helper" required>
-                <div class="field-hint" id="email_address_default_helper">
+                <input class="field-input input-text" type="email" id="email_address_default_helper" name="email_address_default_helper" placeholder="e.g. jane@email.com" aria-describedby="email_address_default_helper_text" required>
```
Comment (talbs) [inline _posts/elements/2015-05-12-forms.md]: Please use ``-`` instead of ``_`` in your value naming conventions.

Comment (talbs) [inline _posts/elements/2015-05-12-forms.md]: Also, please sync up the names your using for helper/hint text. I'd prefer if everything was using one and I'd recommend ``[id]-hint``.

File: _posts/elements/2015-05-12-forms.md [hunk 20]
```diff
@@ -54,8 +54,8 @@ info: Forms allow users to interact with the interface, set preferences, and ent
             </div>
             <div class="field">
                 <label class="field-label field-required" for="email_address_default_suc_helper">Email address (default, has success)</label>
-                <input class="field-input input-text has-success" type="email" id="email_address_default_suc_helper" name="email_address_default_suc_helper" placeholder="e.g. jane@email.com" value="jane@domain.com" aria-describedby="email_address_default_suc_helper" required>
-                <div class="field-hint" id="email_address_default_suc_helper">
+                <input class="field-input input-text has-success" type="email" id="email_address_default_suc_helper" name="email_address_default_suc_helper" placeholder="e.g. jane@email.com" value="jane@domain.com" aria-describedby="email_address_default_suc_helper_text" required>
```
Comment (talbs) [inline _posts/elements/2015-05-12-forms.md]: See [my notes above](https://github.com/edx/ux-pattern-library/pull/126/files#r35094989) and please address.

File: _src/pattern-library/sass/base/_reset.scss (hunk 1 above)
Comment (talbs) [inline _src/pattern-library/sass/base/_reset.scss]: Thanks for the context. I'd prefer a few things:

* Can this use our ``black-t`` and ``white-t`` Sass variables instead of the HTML colors?
* I can see this being used elsewhere in the future, can you make a quick Sass extend in our ``_helpers.scss`` partial and then reference it here and in the ``* {}`` nested rule included below?

File: _src/pattern-library/sass/utilities/_helpers.scss (hunk 2 above)
Comment (talbs) [inline _src/pattern-library/sass/utilities/_helpers.scss]: Hmm, some thoughts:

* What exact value of gray does this fail on? I'd prefer if we don't clobber all links with a darker color just because of one color combo failure
* by using a dark value, we're completely destroying the nice theming relationship a brand primary value had with links/actions. I'd like to find another way to address this case, please.

File: _src/pldoc/static/sass/_components.scss [hunk 21]
```diff
@@ -35,42 +35,46 @@
...
 // ------------------------------
-.pldoc-header-site-title {
-    display: inline-block;
-    vertical-align: middle;
+.pldoc-header-site {
 
-    @include susy-breakpoint($bp-large, $susy) {
-        margin: 0;
-    }
+    .pldoc-header-site-title {
```
Comment (talbs) [inline _src/pldoc/static/sass/_components.scss]: Why did you change this code?

Comment (talbs) [inline _src/pldoc/static/sass/_components.scss]: This should not be nested if its not needed.

File: _src/pldoc/static/sass/_components.scss (hunk 3 above)
Comment (talbs) [inline _src/pldoc/static/sass/_components.scss]: Thanks for the clean up here!

File: _src/pldoc/static/sass/_components.scss [hunk 22]
```diff
@@ -35,42 +35,46 @@
...
-        color: palette(grayscale, white);
-        font-size: font-size(base);
-        font-weight: font-weight(bold);
-        text-transform: uppercase;
-        letter-spacing: rem(1);
+        .title-name {
+            display: block;
+        }
 
-        // STATE: hover, active, focus
```
Comment (talbs) [inline _src/pldoc/static/sass/_components.scss]: Why did you change this code here as well?

**Comment by talbs** [comment]
@clrux, thanks for picking up this work. I've left you a bunch of feedback/changes as well as some questions to address. Let me know when I should jump in again as well as if you want to hash out anything in person.

File: _src/pattern-library/sass/base/_reset.scss (hunk 1 above)
Comment (clrux) [inline _src/pattern-library/sass/base/_reset.scss]: @talbs Do you think using variables is a good idea? This isn't something that can be compromised and if the variables are changed or customized, it could lead to test failures. Granted, this is only used for automated testing - this isn't something that is seen visually. I'll use variables for now, but keep in mind that by using variables you don't necessarily guarantee passing tests.

File: _src/pattern-library/sass/utilities/_helpers.scss (hunk 2 above)
Comment (clrux) [inline _src/pattern-library/sass/utilities/_helpers.scss]: @talbs The gray background is slightly too dark to provide enough contrast with this blue. Some options that might work:
* increase the size of the text
* make it bold
* lighten the gray
* darken the blue

File: _includes/navigation.html (hunk 10 above)
Comment (clrux) [inline _includes/navigation.html]: @talbs Why the parens? This text isn't seen, only announced, and the goal here is to make it read comfortably. "Link, General Styleguide, Link opens in a new window". I'm willing to simplify to just "opens in a new window", but I'd rather not complicate how it's ready with parens (sometimes they're announced, too). Thoughts?

File: _src/pldoc/static/sass/_components.scss (hunk 22 above)
Comment (clrux) [inline _src/pldoc/static/sass/_components.scss]: @talbs Not sure. I didn't add any susy stuff. Maybe it came in from a rebase?

File: _src/pldoc/static/sass/_components.scss (hunk 21 above)
Comment (clrux) [inline _src/pldoc/static/sass/_components.scss]: I don't recall editing this. Maybe it came in from a rebase?

File: _src/pldoc/static/sass/_components.scss (hunk 22 above)
Comment (talbs) [inline _src/pldoc/static/sass/_components.scss]: If its a rebase against the master the diff wouldn't show up. Can you please revert this file/section based on what master has if you don't have any deliberate changes?

File: _includes/navigation.html (hunk 10 above)
Comment (talbs) [inline _includes/navigation.html]: @clrux, I'm thinking of the behavior as a heads up - the content can be read without it, but that note about the new window/tab is supplemental for context purposes. The parens help distinguish the link label from the contextual help. 

Its a lot like "Thai Peanut Chicken with Brown Rice (contains tasty tasty nuts)" vs. "Thai Peanut Chicken with Brown Rice contains tasty tasty nuts.". To me the latter sounds like a bad asian restaurant menu with grammatical errors.

Comment (talbs) [inline _includes/navigation.html]: @cptvitamin, what's the standard here for contextual text?

File: _layouts/default.html (hunk 18 above)
Comment (talbs) [inline _layouts/default.html]: @cptvitamin, is this the standard convention? If so, I'm happy to follow it. Otherwise, I'd prefer that we rely on the semantic value of ``<main>`` by itself.

File: _src/pldoc/static/sass/_components.scss (hunk 21 above)
Comment (talbs) [inline _src/pldoc/static/sass/_components.scss]: Did you rebase, amigo?

File: _src/pldoc/static/sass/_components.scss (hunk 22 above)
Comment (talbs) [inline _src/pldoc/static/sass/_components.scss]: Did you rebase, amigo?

**Comment by clrux** [comment]
Having rebase woes. Going to close this branch and make a new one, cherrypicking in the work I did. Standby.
